[10.0][MIG] l10n_es_partner

pedrobaeza opened the pull request:
WIP hasta que se fusione
- [x] #405 

cc @Tecnativa

pedrobaeza committed:
l10n_es_partner

carlosdauden committed:
l10n_es_partner: Return commercial name in name_get with configurable pattern

mileo reviewed (approved):
LGTM :+1: no test.

mileo commented:
Descargar el excel de las 'Entidades con establecimiento' de la web del Banco de España: http://goo.gl/EPfcR5 2. Mover el archivo descargado 'REGBANESP_CONESTAB_A.XLS' a la carpeta gen_src 3. Ejecutar

This google link isn't working anymore.

pedrobaeza committed:
l10n_es_partner

* Better search on partner
* Review fixes
* More tests

pedrobaeza committed:
l10n_es_partner: Migration script

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
l10n_es_partner: Improve search over other terms

When searching on the omnibar, you can include other filters and
the default search includes other different arguments, so the
condition for adding the commercial name should be expanded.

Roodin commented:
Me ha dado errores al arrancar por problemas en los import:
l10n_es_partner/models/res_partner.py
from odoo import expression
y 
l10n_es_partner/models/res_partner_bank.py
from odoo.addons.base_iban.base_iban import

Y he tenido que modificar las rutas de estos import para ponerlo en marcha

pedrobaeza commented:
OK, reviso. En realidad, como estaba pendiente de primero mergear el l10n_es_toponyms, a éste todavía no le había prestado mucha atención. Esta tarde estoy montando una v10, así que depuraré esas cosas.

pedrobaeza commented:
@Roodin ya está revisado el módulo. Puedes probar ahora. He incluido también los commits de l10n_es_toponyms para que se pueda probar en runbot.

Roodin commented:
Ok. Ya lo había probado cambiando, pero volveré a hacerlo

El 20 ene. 2017 8:57 p. m., "Pedro M. Baeza" <notifications@github.com>
escribió:

> @Roodin <https://github.com/Roodin> ya está revisado el módulo. Puedes
> probar ahora. He incluido también los commits de l10n_es_toponyms para que
> se pueda probar en runbot.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/l10n-spain/pull/411#issuecomment-274165266>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKpVC3gaYLTjBPTrdgLno6-u76FBBBMDks5rURHCgaJpZM4LJhgD>
> .
>

pedrobaeza commented:
@mileo, he puesto el enlace directo sin acortador para que no haya problema.

pedrobaeza committed:
[IMP] l10n_es_partner: Make name_search work

As Odoo overwrites name_search on res.partner in a non inheritable way,
we have to include explicitly name_search method and search first for
the partners with that commercial, and then the rest.

pedrobaeza committed:
[FIX] l10n_es_partner: fix for name_search when limit=None

limit parameter con be 0 or None to representate infinite, so it
should be handled.

angelmoya commented:
Probado en instalación local, con importación de fichero local y funciona perfectamente

felixvillafranca commented:
Con importación desde GeoNames y Bancos desde fichero funciona correctamente ¡¡¡

angelmoya commented:
Hola @pedrobaeza  he encontrado un error, replicable en runbot.

Ve a Contacts, y en el buscador pon un nombre y busca por "Related Company "

Te da este error:

```
  File "/.repo_requirements/odoo/odoo/osv/expression.py", line 1052, in parse
    push(create_substitution_leaf(leaf, _get_expression(comodel, left, right, operator), model))
  File "/.repo_requirements/odoo/odoo/osv/expression.py", line 1045, in _get_expression
    res_ids = [x[0] for x in comodel.with_context(active_test=False).name_search(right, [], operator, limit=None)]
  File "/root/build/OCA/l10n-spain/l10n_es_partner/models/res_partner.py", line 45, in name_search
    limit_rest = limit - len(partners)
TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'
```

pedrobaeza commented:
¿Eso lo has reproducido con la última versión?

angelmoya commented:
Hace 10 minutos en el runbot recién hecho

El 18/3/2017 13:58, "Pedro M. Baeza" <notifications@github.com> escribió:

> ¿Eso lo has reproducido con la última versión?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/l10n-spain/pull/411#issuecomment-287544795>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGcAhYAT3vBpCMgpTHucnDHCn0dGuXbJks5rm9TXgaJpZM4LJhgD>
> .
>

pedrobaeza commented:
OK, miro a ver.

pedrobaeza committed:
[MIG] l10n_es_partner: Migration to 10.0

pedrobaeza commented:
Vale, el problema era que no había incluido el último commit de la 9.0 que lo repara: 1f06845f66f35689a236643866cd23068e41fce4

Ya está incluido.

angelmoya commented:
OK, probado en runbot y ya va bien

:+1:

pedrobaeza commented:
Fusiono entonces ya.
